### PR TITLE
feat(dbt-fal): make dbt-fal work with sqlserver

### LINKED ISSUE #778: Support sqlserver
Add support for MS sqlserver

### PR DESCRIPTION
**Pull request by mederka**
## Description
Adds support for sqlserver adapter
Closes #778

### CODE REVIEW

**Comment by mederka** [comment]
@chamini2, should we add integration tests for sqlserver? Not sure about the overhead.

**Comment by chamini2** [comment]
> @chamini2, should we add integration tests for sqlserver? Not sure about the overhead.

I would say we should @mederka. We can later make some tests run only in certain cases, but having the integration tests seems important if we have sqlserver code

File: .github/workflows/test_integration_adapter.yml [hunk 1]
```diff
@@ -32,6 +32,7 @@ jobs:
           - snowflake
           - redshift
           - trino
+          # - sqlserver # does not support dbt 1.4 yet
```
Comment (chamini2) [inline .github/workflows/test_integration_adapter.yml:35]: OK, this is interesting for me. We deploy latest from this branch. Should this be pointing to a [1.3 branch](https://github.com/fal-ai/fal/tree/release-adapter-1.3)?

Comment (chamini2) [reply .github/workflows/test_integration_adapter.yml:35]: Or just wait for 1.4?

Comment (mederka) [reply .github/workflows/test_integration_adapter.yml:35]: I think we can just wait.

**Review by chamini2 — APPROVED** [review]

**Pull request merged**